src/seedu/addressbook/AddressBook.java

```diff
@@ -986,18 +1004,31 @@ private static String extractNameFromPersonString(String encoded) {
     private static String extractPhoneFromPersonString(String encoded) {
```

Comment:
Seems like this method is too large.
Could `SLAP` be used here to help the code be more readable?